tools/pipelines/build-docs.yml

```diff
@@ -88,39 +88,54 @@ variables:
   - name: pnpmStorePath
     value: $(Pipeline.Workspace)/.pnpm-store
 
+# TODO: include release/* branches as trigger. Currently, the website docs builds and deploys on each commit.
+# However, inconsistencies between the release branches and main is causing the website to change unexpectedly
+# we can re include the release/* branch to the trigger after separating the deploy step to another pipeline
 # no PR triggers
 trigger:
   branches:
     include:
-    - release/*
+    # - release/*
     - main
 pr: none
 
+# TODO: the download artifact step is currently targetting the latest pipeline run from each resource.
+# However, this is causing inconsistencies in the doc in scenarios where the latest runs on each pipeline
+# is not the same (e.g. mix of main and lts). Temporarily setting the branch to main explicitly.
 resources:
   pipelines:
   - pipeline: common_definitions
     source: Build - common-definitions
+    branch: main
```

Comment:
One note here: this is definitely an improvement, because we will only trigger the run of _this pipeline_ when the specified build pipelines run _on main only_ complete. But there is still a potential race condition here. Namely, each run of `build-npm-package.yaml` (used by each of these build-foo pipelines) will publish a new version of their corresponding doc model artifact, regardless of what branch the pipeline is run from. So one of those pipelines could still be running right before this one does and publish a new (invalid/obsolete) artifact that this pipeline will end up consuming. 

We likely need to change the various build pipelines to only specify the `publishDocs` parameter when they are being run from `main` (rather than always specifying true). This could be done in a separate PR though - what you have here is still an improvement on its own.

Reply:
Let me know if that doesn't make sense. There's a lot going on here that we need to straighten out, it looks like 😕.

Reply:
Another option would be to update `build-npm-package` to include some info about the branch it was run from in the artifact it publishes. Then this pipeline could strictly look at the artifacts generated for the particular branches it cares about (for now: main). You're going to need to do something like this eventually to make the website use the artifact from the v2 release branches instead of main anyways, but it doesn't strictly need to be done now.

Reply:
> because we will only trigger the run of this pipeline when the specified build pipelines run on main only complete.

I don't think that's correct. This only says which completed run of that pipeline to use when this one triggers (for any reason). There's a separate `trigger` section for each `resources.pipelines.pipeline` that controls whether a new pipeline run of that "dependency" resource triggers this one ([docs](https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/resources-pipelines-pipeline?view=azure-pipelines)). Today, AFAICT, runs of all the "dependencies" declared here do not cause build-docs to trigger.

I think updating the branch from which the pipeline downloads artifacts in the download steps (what I described [here](https://github.com/microsoft/FluidFramework/pull/19190#discussion_r1447945331)) might accomplish what you want, though.